qcore/gmt.py

```diff
@@ -2421,6 +2426,8 @@ def clip(self, path=None, is_file=False, invert=False, n=None):
             cmd = [GMT, "psclip", "-J", "-R", "-K", "-O", self.z]
             if invert:
                 cmd.append("-N")
+            if self.p:
```

Comment:
Can you explain why perspective option is handled in this function, and how this is related to the problem you are trying to solve?

Reply:
in __init__
# perspective mode, 3D mode default
self.p = False

It's a bool that determines if the plot has perspective.